TebLocalPlannerROS:

 odom_topic: odom
 map_frame: map
    
 # Trajectory
  
 teb_autosize: True
 dt_ref: 0.3      # 最优路径上的两个相邻姿态（即位置、速度、航向信息，可通过TEB可视化在rivz中看到）的默认距离
 dt_hysteresis: 0.1
 max_samples: 500
 global_plan_overwrite_orientation: False  #覆盖由全局规划器提供的局部子目标的方向. 对于车辆的2D规划，可以设置为False，可实现对全局路径的更好跟踪。
 allow_init_with_backwards_motion: true
 max_global_plan_lookahead_dist: 8.0   #  最大向前看距离
 global_plan_viapoint_sep: -1     # 全局规划路径中提取的路点的相互距离
 global_plan_prune_distance: 1
 exact_arc_length: False
 feasibility_check_no_poses: 2
 publish_feedback: False

 force_reinit_new_goal_dist: 0.0   # ?

 # Robot

 max_vel_x: 1.0     # 速度约束
 max_vel_x_backwards: 0.0     # 最大倒车速度
 max_vel_y: 0.0   # y轴方向速度
 max_vel_theta: 1.0 # 角速度 the angular velocity is also bounded by min_turning_radius in case of a carlike robot (r = v / omega)
 acc_lim_x: 1.0     # 加速度约束
 acc_lim_theta: 0.5    # 角加速度约束

 # ********************** Carlike robot parameters ********************
 min_turning_radius: 0        # Min turning radius of the carlike robot (compute value using a model or adjust with rqt_reconfigure manually)
 wheelbase: 0.498               # Wheelbase of our robot
 cmd_angle_instead_rotvel: false # stage simulator takes the angle instead of the rotvel as input (twist message)
 # ********************************************************************

 footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
   type: "polygon"
   radius: 0.2 # for type "circular"
   line_start: [-0.1, 0.0] # for type "line"
   line_end: [0.1, 0.0] # for type "line"
   front_offset: 0.2 # for type "two_circles"
   front_radius: 0.2 # for type "two_circles"
   rear_offset: 0.2 # for type "two_circles"
   rear_radius: 0.2 # for type "two_circles"
   vertices: [[-0.5, -0.4], [-0.5, 0.4], [0.5, 0.4], [0.5, -0.4]] # for type "polygon"

 # GoalTolerance
    
 xy_goal_tolerance: 0.5    # 目标跟踪误差
 yaw_goal_tolerance: 0.3    # 目标跟踪误差
 free_goal_vel: true   # 自由目标速度,设为False时，车辆到达终点时的目标速度为0
 complete_global_plan: false

 # Obstacles

 min_obstacle_dist: 0.1 # This value must also include our robot's expansion, since footprint_model is set to "line".
 inflation_dist: 0.0
 include_costmap_obstacles: True
 costmap_obstacles_behind_robot_dist: 1.0
 obstacle_poses_affected: 15

 dynamic_obstacle_inflation_dist: 0.0
 include_dynamic_obstacles: True 

 costmap_converter_plugin: ""
 costmap_converter_spin_thread: True
 costmap_converter_rate: 5

 # Optimization
    
 no_inner_iterations: 5
 no_outer_iterations: 4
 optimization_activate: True
 optimization_verbose: False
 penalty_epsilon: 0.5
 obstacle_cost_exponent: 4
 weight_max_vel_x: 10000
 weight_max_vel_theta: 1
 weight_acc_lim_x: 1
 weight_acc_lim_theta: 1
 weight_kinematics_nh: 1000
 weight_kinematics_forward_drive: 100
 weight_kinematics_turning_radius: 1
 weight_optimaltime: 1 # must be > 0
 weight_shortest_path: 1
 weight_obstacle: 200
 weight_inflation: 0.2
 weight_dynamic_obstacle: 10 # not in use yet
 weight_dynamic_obstacle_inflation: 0.2
 weight_viapoint: 10
 weight_adapt_factor: 2

 # Homotopy Class Planner

 enable_homotopy_class_planning: True
 enable_multithreading: True
 max_number_classes: 4
 selection_cost_hysteresis: 1.0
 selection_prefer_initial_plan: 0.95
 selection_obst_cost_scale: 1.0
 selection_alternative_time_cost: False

 roadmap_graph_no_samples: 15
 roadmap_graph_area_width: 5
 roadmap_graph_area_length_scale: 1.0
 h_signature_prescaler: 0.5
 h_signature_threshold: 0.1
 obstacle_heading_threshold: 0.45
 switching_blocking_period: 0.0
 viapoints_all_candidates: True
 delete_detours_backwards: True
 max_ratio_detours_duration_best_duration: 3.0
 visualize_hc_graph: False
 visualize_with_time_as_z_axis_scale: False

# Recovery
 
 shrink_horizon_backup: True
 shrink_horizon_min_duration: 10
 oscillation_recovery: True
 oscillation_v_eps: 0.1
 oscillation_omega_eps: 0.1
 oscillation_recovery_min_duration: 10
 oscillation_filter_duration: 10
